src/middleware/socketioMiddleware.js

```diff
@@ -43,6 +48,18 @@ function socketioMiddleware() {
         if (previousTeam) leaveTeam(previousTeam);
         joinTeam(action.team);
       }
+
+      if (action.type === SYNCHRONIZED_SEARCH) {
```

Comment:
Maybe use a switch here, or at least return after each if so it doesn't check the others unnecessarily.